### ISSUE TITLE
Stress-testing the checking of broken links to headings

### ISSUE THREAD

**Issue by claremacrae** [ipshing/obsidian-broken-links#1]
Very many thanks for this plugin. I've put it through its paces today, and fixed many broken links to headings.

There were a few that were actually working OK in Obsidian - it navigated to the heading... but the plugin thinks the heading is broken.

They are all corner cases, but in case you are interested, I thought I would pop the examples in an issue.

- Obsidian 1.5.5
- Dangling Links 1.0.4

### Steps to reproduce

1. Create a note called `Heading links that Broken Links says are broken but work`
2. Shift+Paste in this content:

```text
---
creation date: 2024-02-14 17:39
tags: 
aliases:
  -
---

# Heading links that Broken Links says are broken but work

Dangling links 1.0.4 warns that the following headings are all broken links.

They all work in Obsidian 1.5.5.

### Can I link to a heading that contains a wikilink to [[Heading links that Broken Links says are broken but work]]

- [[#Can I link to a heading that contains a wikilink to Heading links that Broken Links says are broken but work]]
- Obsidian removes the `[[` and `]]` characters in the heading, when linking to this heading.
- But the link 

### Can I link to a heading that ends with a question mark?

- [[#Can I link to a heading that ends with a question mark]]
- Earlier versions of Obsidian used to remove punctuation when selecting heading names in links...
- This link still works

### Can I link to a heading that contains a comma, somewhere in it

- [[#Can I link to a heading that contains a comma somewhere in it]]
- Obsidian 1.5.5 does not include the colon (`:`) when linking to this heading.

### PR #2015 - feat: Toggle task done command adds global filter text, if enable

- [[#PR 2015 - feat Toggle task done command adds global filter text, if enable]]
- This link was generated by Obsidian auto-complete - and Broken links shows an error
- I believe it is because Obsidian did not include the `#` in `#2015` in the link.
```
3. View the file's broken links in the Broken Links panel:

**Expected Result**

- No broken links

**Actual Result**

- This:
    ![image](https://github.com/ipshing/obsidian-broken-links/assets/4840096/c1381298-665e-4991-aed0-bada14a9f9dc)

**Comment by ipshing** [comment]
I was not properly comparing against file headings, this should be fixed in [1.0.5](https://github.com/ipshing/obsidian-broken-links/releases/tag/1.0.5).

**Comment by claremacrae** [comment]
Confirmed that all cases are fixed - many thanks!
